### ISSUE TITLE
counting reads in CRAM files fails without references

### ISSUE THREAD

**Issue by vadimzalunin** [samtools/samtools#628]
Not sure if this is a bug or a feature:

> samtools view -c DRR003152.cram
Failed to populate reference for id 0
Unable to fetch reference #0 147436..412745
Failure to decode slice
[main_samview] truncated file.

Version: 1.3.1 (using htslib 1.3.1)
The file: ftp://ftp.sra.ebi.ac.uk/vol1/ERZ359/ERZ359910/DRR003152.cram

**Comment by jkbonfield** [comment]
What do you have $REF_PATH set to?

I get this error if I set REF_PATH to a local directory *only* without having that reference cached in the local directory.  If REF_PATH also contains the EBI reference server, it checks the first directory (fails) and then downloads it from EBI.  If you unset REF_PATH completely this also happens.

The only time I could get the error is by setting REF_PATH explicitly and not including the reference server as one of the components.  That's working correctly I think.

**Comment by jkbonfield** [comment]
That said, I agree that it ought to be possible to count the number of reads without knowing what the reads are.  This is infact supported, but is absent in the view -c logic.

Eg try:

```
samtools view --input-fmt-option required_fields=flag -c ftp://ftp.sra.ebi.ac.uk/vol1/ERZ359/ERZ359910/DRR003152.cram TGACv1_scaffold_000001_1AL
```

This gives me 306 even after deleting the local copy of the reference.  Without the ```--input-fmt-option required_fields=flag``` it attempts to work out the sequence and fails.  Therefore this is a lost optimisation opportunity in view -c.

**Comment by vadimzalunin** [comment]
thanks, --input-fmt-option is useful!

**Comment by jkbonfield** [comment]
Closing as apparently we finally fixed this in 1.17.  Bonus
